### ISSUE TITLE
[Feature Request] DB Queues 

### ISSUE THREAD

**Issue by BlackYuzia** [alliedmodders/sourcemod#1724]
### System

**OS**: Ubuntu 16.04, 20.04
**SM**: 1.10
**MM**: latest

### Description

Hi there.

How we know, we have 2 database drivers (SQLite and MYSQL). In 1.11+ we also have third driver - PostgreSQL.
**How I know**, they work in one (or single) **queue**.

If they have one queue, that also give us a bug / feature / side effect: when mysql disconnect by some reasons, I suppose, what MYSQL driver just block all queue by connection(s) and, maybe, queries. Because we have one queue for MYSQL and SQLite, all SQLIte queries all stop and wait until MYSQL complete they connections and etc. If MYSQL can't connecct, that repeat infinity times...

### Reproduce

To reproduce:
1. You sould have a MYSQL and SQLite databases (plugins what get / set data inside and from SQLIte and MYSQL databases)
2. **When your server works and have connection with MYSQL database**, then database (host) should drop / lost connection and do not connect again **never** (by some reasons, in my case - too bad ethernet or lost him on database host by WAR in Ukraine, maybe you could get the same result?)
3. MYSQL should send many try connections  to database and block queue.
4. SQLIte would just wait until he can get data, but he don't get data until MYSQL driver get connect from database.
5. Success! MYSQL driver was block queue and SQLIte driver can't get any data because queue is blocked by MYSQL!

### Feature Request 

As I say, with single databases drivers queue we have bug(s), so my request: Add one queue per one driver (SQLite and MYSQL).

That should fix a bug with blocking queue by MYSQL Driver.

**Comment by VOTOSH** [comment]
This Ukrainian pig is a liar,

**Comment by VOTOSH** [comment]
I have the same problem on my project.

**Comment by asherkin** [comment]
See previously: #1076 (which looks like it was closed accidentally.)

Thread-per-driver is a slightly odd way to split this up, I'd previously looked at options around thread-per-connection (or per-plugin) but touching the DBI threading model is extremely brittle until the work outlined in #1207 is complete - which likely opens us up to solving the underlying problem better by keeping the single worker thread and switching to async APIs under the hood..

**Comment by Alexeyt89** [comment]
I have the same problem, when I reload a plugin which has unfinished threaded queries, the queries are turned into non-threaded and if the connection to the MySQL server is not stable, the plugin locks the thread which is eventually killed by watchdog.

	/* Mark the plugin as being unloaded so future database calls will ignore threading... */
	plugin->SetProperty("DisallowDBThreads", NULL);
	
	/* Run all of the think operations.
	* Unlike the driver unloading example, we'll let these calls go through,
	* since a plugin unloading is far more normal.
	*/

I see two ways here - detache all pending queries from the plugin and keep them threaded, or drop them. The information in the queries can't be more important than server stability.
